### ISSUE TITLE
Maniac Mansion Mania 066: "First Time Enter" room event not called in room 1

### ISSUE THREAD

**Issue by tag2015** [adventuregamestudio/ags#2132]
**Describe the bug**
Maniac Mansion Mania 066 fails a cutscene check when opening the front door after the start of the game:
 !StartCutscene: already in a cutscene; previous started in "Room 24 script", line 14!

**AGS Version**
3.6.1.7 beta8
3.6.0.51

**Game**
[Download](https://maniac-mansion-mania.com/index.php/en/spiele/episoden/staffel-7/513-episode-66-hoagies-neuer-freund.html)

**To Reproduce**
1, Watch intro (can click to skip texts)
2. Open door
3. Go downstairs
4. Open leftmost door

**Desktop (please complete the following information):**
 - Win 10

**Additional context**
Additional bug: hitting ESC to bypass the intro freezes the game.

**Comment by ivan-mogilko** [comment]
For the reference, original engine also errors on starting a "nested" cutscene. Which makes me wonder if the engine failed to register the ending of the previous cutscene.

**Comment by ivan-mogilko** [comment]
So, after playing original game for comparison, I found there's another mistake, which may be also related.

After skipping the initial intro and appearing in the character's bedroom, Hoagie character sais several speech lines in the original game. However, the current engine does not display these lines.

**Comment by AlanDrake** [comment]
This looks a change in behavior from 2.7 and the earliest code CJ made available to us.

I was curious if simply not throwing that error would work, but it complains about the following with an acwin compiled on master branch:
```
Error: Error running function 'game_start':
Error (line 1196): ScriptGame: attempt to write readonly variable at offset 232

in the active script:
in "Global script", line 1196
```

**Comment by ivan-mogilko** [comment]
I fixed this assertion with the commit: 09a13f8dc0f2fc9e58f5eb0103066197e598cb73.

It's possible to ignore the nested StartCutscene, but like I mentioned in the above comment, the game is stuck with "in cutscene" flag set, and some speech lines are not played in the first room, possibly for the related reason.

**Comment by ivan-mogilko** [comment]
Did a quick update to our lazily done crmpak tool to let unpack and decode the room scripts from ags 2 games: 75763bb183a415cb5fcbc91a01521143ba5ce71c

Here's the part of the starting room 1 that should end cutscene and display speech lines:
```
#sectionstart room_b  // DO NOT EDIT OR REMOVE THIS LINE
function room_b() {
  // script for Room: First time player enters room
EndCutscene();
player.Say("*Gдhn*");  
player.Say("Das Leben ist ungerecht.");
player.Say("Dad ist auf einer Dichterlesung...");
player.Say("... der Kьhlschrank ist leer...");
player.Say("... und ich bin zu faul um einkaufen zu gehen.");
player.Say("Vielleicht sollte ich versuchen mich einen Tag ьber von Luft zu ernдhren.");
player.Say("Eine Sauerstoff-Diдt oder sowas.");
cMet.ChangeRoom(12,103,129);
}
#sectionend room_b  // DO NOT EDIT OR REMOVE THIS LINE
```

I begin to wonder, if the new engine simply skips this function "room_b" for some reason. Need to research if these are run at all now (they could have been hardcoded to run previously).

**Comment by ivan-mogilko** [comment]
@tag2015 I pushed a possible fix: 45927bf23ea4ca051f74154fa4d0561a7df379ed

This fixes "first time enter room" event *not* being called if the room change was ordered from within a old-format dialog script (i think).

**Comment by tag2015** [comment]
> @tag2015 I pushed a possible fix: [45927bf](https://github.com/adventuregamestudio/ags/commit/45927bf23ea4ca051f74154fa4d0561a7df379ed)
> 
> This fixes "first time enter room" event _not_ being called if the room change was ordered from within a old-format dialog script (i think).

Thanks, the fix seems to work
